components/calendar/examples/iso_datetime_manipulations.rs

```diff
@@ -0,0 +1,64 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// An example application which uses icu_datetime to format entries
+// from a log into human readable dates and times.
+
+#![no_main] // https://github.com/unicode-org/icu4x/issues/395
+
+icu_benchmark_macros::static_setup!();
+
+use icu_calendar::{DateTime, DateTimeError, Iso};
+
+const DATETIMES_ISO: &[(i32, u8, u8, u8, u8, u8)] = &[
+    (1970, 1, 1, 3, 5, 12),
+    (1982, 3, 11, 2, 25, 59),
+    (1999, 2, 21, 13, 12, 23),
+    (2000, 12, 29, 10, 50, 23),
+    (2001, 9, 8, 11, 5, 5),
+    (2017, 7, 12, 3, 1, 1),
+    (2020, 2, 29, 23, 12, 23),
+    (2021, 3, 21, 18, 35, 34),
+    (2021, 6, 10, 13, 12, 23),
+    (2021, 9, 2, 5, 50, 22),
+    (2022, 10, 8, 9, 45, 32),
+    (2022, 2, 9, 10, 32, 45),
+    (2033, 6, 10, 17, 22, 22),
+];
+
+fn print(_input: &str) {
+    #[cfg(debug_assertions)]
+    println!("{}", _input);
+}
+
+fn tuple_to_iso_datetime(date: (i32, u8, u8, u8, u8, u8)) -> Result<DateTime<Iso>, DateTimeError> {
+    DateTime::new_iso_datetime_from_integers(date.0, date.1, date.2, date.3, date.4, date.5)
+}
+
+#[no_mangle]
+fn main(_argc: isize, _argv: *const *const u8) -> isize {
+    icu_benchmark_macros::main_setup!();
+
+    let datetimes = DATETIMES_ISO
+        .iter()
+        .copied()
+        .map(tuple_to_iso_datetime)
+        .collect::<Result<Vec<DateTime<Iso>>, _>>()
+        .expect("Failed to parse datetimes.");
+
+    for datetime in datetimes.iter() {
+        let formatted_datetime = format!(
+            "Year: {}, Month: {}, Day: {}, Hour: {}, Minute: {}, Second: {}",
+            datetime.date.year().number,
+            datetime.date.month().number,
+            datetime.date.day_of_month().0,
+            u8::from(datetime.time.hour),
+            u8::from(datetime.time.minute),
+            u8::from(datetime.time.second),
```

Comment:
same here - make `print` accept the struct that has the right components, and conditionally print them. This way you format the items but don't allocate into String as part of the measurement.